src/service-worker.js

```diff
@@ -169,8 +171,14 @@ self.addEventListener('fetch', event => {
 self.addEventListener('push', event => {
   event.waitUntil((async () => {
     const data = event.data.json()
-    const { origin } = event.target
+    const knownInstances = await getKnownInstances()
+    if (knownInstances.length !== 1) {
+      // TODO: find a way to determine the pushing instance from the PushEvent
+      await showSimpleNotification(data)
+      return
+    }
 
+    const origin = basename(knownInstances[0])
```

Comment:
This is an unfortunate solution because a user might log into 2 instances and enable notifications on the second one.

It seems like a better system would be for us to log somewhere _which_ instance they've enabled notifications for. I.e. we can enforce the "you can only have notifications for one instance at once time" system ourselves.

Reply:
Indeed it's not ideal. For the case where a user logged into 2 instances, they'd have "simple" notifications even if they did enable notifications only on one of them.

Note that this line gets the *single* element from `knownInstances`, as the case where there are more than one (more accurately, not exactly one) is handled above; so it's not a matter of "enabling notifications on the second one".